### PR TITLE
Frogger Game is ready for review

### PR DESCRIPTION
**Pull request by m-ruslan**
### Frogger Game

[Demo](https://m-ruslan.github.io/jsoop-practice/) |
[Code base](https://github.com/m-ruslan/jsoop-practice)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/m-ruslan/js-oop-practice/app.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+var pDefualtYPos = rowStep * 5 - 30;
+
+var player = new Player(pDefualtXPos, pDefualtYPos);
+
+var enemy1 = new Enemy(-colStep, rowStep * 3 - 30, 100);
+var enemy2 = new Enemy(-colStep, rowStep * 2 - 30, 200);
+var enemy3 = new Enemy(-colStep, rowStep * 1 - 30, 300);
+var enemy4 = new Enemy(-1000 - colStep, rowStep * 3 - 30, 200);
+var enemy5 = new Enemy(-1000 - colStep, rowStep * 2 - 30, 250);
+var enemy6 = new Enemy(-1000 - colStep, rowStep * 1 - 30, 300);
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/js-oop-practice/app.js:118]: Quite many repetitions. Can you devise an approach to benefit from calculable values, dependence on enemy index and eliminating repetitons?
Imagine we need to create dozens of enemies.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@m-ruslan well done!

Below is the checklist focused on the room for improvements in the code. Please also check comments below.

 * [ ] the code is very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
 * Requirements re **Constants**:
   * [ ] all numbers like block dimensions, initial locations are defined as named constants
         (e.g. `const STEP = 101;`) as otherwise numbers scattered across code base look cryptic;
         named constants add semantic meaning and improve readability
   * [ ] there are core constants and derived constants
         (e.g. derived constant `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants 
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`
 * Requirements re **OOP**:
   * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
         (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
         Hint: pass `Player` instance as an argument to every enemy
   * [ ] Separation of Concerns principle is followed
         (e.g. `update` method does only rendering and doesn't contain any unrelated **inline** code;
         for example collision check is defined as a dedicated method and only called from inside `update`)

File: submissions/m-ruslan/js-oop-practice/app.js (hunk 1 above)
Comment (m-ruslan) [reply submissions/m-ruslan/js-oop-practice/app.js:118]: @OleksiyRudenko thank you
I'll refactor my code and will come back to you for one more review)

**Comment by m-ruslan** [comment]
@OleksiyRudenko  it seems to me that I've fixed all issues you wrote about. 
Could you please review?

**Review by OleksiyRudenko — APPROVED** [review]
@m-ruslan great job!

**Pull request merged**